iset.mm

```diff
@@ -135016,6 +135038,642 @@ the base set to the (extended) reals and which is nonnegative, symmetric
+      ( vf cv cds cfv cbs cxp cres cmet wcel cxms cms fveq2 syl6eqr wceq fveq2d
+      sqxpeqd reseq12d eleq12d df-ms elrab2 ) HIZJKZUHLKZUJMZNZUJOKZPADOKZPHCQR
+      UHCUAZULAUMUNUOULCJKZDDMZNAUOUIUPUKUQUHCJSUOUJDUOUJCLKDUHCLSFTZUCUDGTUOUJ
+      DOURUBUEHUFUG $.
+
+    $( Express the predicate " ` <. X , D >. ` is a metric space" with
+       underlying set ` X ` and distance function ` D ` .  (Contributed by NM,
+       27-Aug-2006.)  (Revised by Mario Carneiro, 24-Aug-2015.) $)
+    isms2 $p |- ( K e. MetSp <->
+        ( D e. ( Met ` X ) /\ J = ( MetOpen ` D ) ) ) $=
+      ( cxms wcel cmet cfv wa cxmet cmopn wceq cms isxms2 anbi1i isms metxmet
+      pm4.71ri an32 bitri 3bitr4i ) CHIZADJKIZLADMKIZBANKOZLZUFLZCPIUFUHLZUEUIU
+      FABCDEFGQRABCDEFGSUKUGUFLZUHLUJUFULUHUFUGADTUARUGUFUHUBUCUD $.
+
+    $( The topology component of a metric space coincides with the topology
+       generated by the metric component.  (Contributed by Mario Carneiro,
+       26-Aug-2015.) $)
+    xmstopn $p |- ( K e. *MetSp -> J = ( MetOpen ` D ) ) $=
+      ( cxms wcel ctps cmopn cfv wceq isxms simprbi ) CHICJIBAKLMABCDEFGNO $.
+
+    $( The topology component of a metric space coincides with the topology
+       generated by the metric component.  (Contributed by Mario Carneiro,
+       26-Aug-2015.) $)
+    mstopn $p |- ( K e. MetSp -> J = ( MetOpen ` D ) ) $=
+      ( cms wcel cmet cfv cmopn wceq isms2 simprbi ) CHIADJKIBALKMABCDEFGNO $.
+  $}
+
+  $( A metric space is a topological space.  (Contributed by Mario Carneiro,
+     26-Aug-2015.) $)
+  xmstps $p |- ( M e. *MetSp -> M e. TopSp ) $=
+    ( cxms wcel ctps ctopn cfv cds cbs cxp cres cmopn wceq eqid isxms simplbi )
+    ABCADCAEFZAGFAHFZQIJZKFLRPAQPMQMRMNO $.
+
+  $( A metric space is a topological space.  (Contributed by Mario Carneiro,
+     26-Aug-2015.) $)
+  msxms $p |- ( M e. MetSp -> M e. *MetSp ) $=
+    ( cms wcel cxms cds cfv cbs cxp cres cmet ctopn eqid isms simplbi ) ABCADCA
+    EFAGFZOHIZOJFCPAKFZAOQLOLPLMN $.
+
+  $( A metric space is a topological space.  (Contributed by Mario Carneiro,
+     26-Aug-2015.) $)
+  mstps $p |- ( M e. MetSp -> M e. TopSp ) $=
+    ( cms wcel cxms ctps msxms xmstps syl ) ABCADCAECAFAGH $.
+
+  ${
+    msf.x $e |- X = ( Base ` M ) $.
+    msf.d $e |- D = ( ( dist ` M ) |` ( X X. X ) ) $.
+    $( The distance function, suitably truncated, is a metric on ` X ` .
+       (Contributed by Mario Carneiro, 2-Sep-2015.) $)
+    xmsxmet $p |- ( M e. *MetSp -> D e. ( *Met ` X ) ) $=
+      ( cxms wcel cxmet cfv ctopn cmopn wceq eqid isxms2 simplbi ) BFGACHIGBJIZ
+      AKILAPBCPMDENO $.
+
+    $( The distance function, suitably truncated, is a metric on ` X ` .
+       (Contributed by Mario Carneiro, 12-Nov-2013.) $)
+    msmet $p |- ( M e. MetSp -> D e. ( Met ` X ) ) $=
+      ( cms wcel cmet cfv ctopn cmopn wceq eqid isms2 simplbi ) BFGACHIGBJIZAKI
+      LAPBCPMDENO $.
+
+    $( Mapping of the distance function of a metric space.  (Contributed by NM,
```

Comment:
The distance function of a metric space is a function into the real numbers.

Reply:
Proposal adopted in set.mm and iset.mm.